core/base/surfaceGeometrySmoother/SurfaceGeometrySmoother.h

```diff
@@ -146,6 +134,39 @@ namespace ttk {
                      const triangulationType0 &triangulationToSmooth,
                      const triangulationType1 &triangulationSurface) const;
 
+    /**
+     * @brief Find nearest vertex on the surface.
+     *
+     * @param[in] pa Input point
+     * @param[in] dists Pre-allocated distances vector
+     * @param[in] triangulation Surface triangulation
+     * @return Nearest vertex id on the surface
+     */
+    template <typename triangulationType>
+    inline SimplexId
+      getNearestSurfaceVertex(const Point &pa,
```

Comment:
I believe this function should be moved to the geometry class (since it is now used by multiple classes and since that call is more generic that the use case of surface geometry smoothing).

Reply:
The `Point` structure is declared in `SurfaceGeometrySmoother.h`.
If I move `getNearestSurfaceVertex` to geometry I can either:
- move the `Point` structure to geometry
- Change the signature of `getNearestSurfaceVertex` to not use the `Point` structure (because not used in geometry for now)

Reply:
Done in the next commit!